DDF-3738 Move "Add Visual" menu to be in-line with other action menu options

alexaabrd opened the pull request:
#### What does this PR do?
Moves "Add Visual" menu to be in-line with other action menu options

#### Who is reviewing it? 
@lcrosenbu 
@idperez 
@mackncheesiest 
@Variadicism 
@SmithJosh 
@vinamartin 

#### Select relevant component teams: 
https://github.com/orgs/codice/teams/ui
#### Choose 2 committers to review/merge the PR. 
@andrewkfiedler
@brendan-hofmann
#### How should this be tested? (List steps with links to updated documentation)
Build branch & Startup ddf
test '[+]' functionality in workspace, upload and alert views
#### Any background context you want to provide?
#### What are the relevant tickets?
[DDF-3738](https://codice.atlassian.net/browse/DDF-3738)
#### Screenshots (if appropriate)
mockup:
![image](https://user-images.githubusercontent.com/14789712/38595057-66337eb8-3cfe-11e8-8db5-a990477e17e5.png)
actual:
![image](https://user-images.githubusercontent.com/14789712/39845749-817ea71e-53ac-11e8-8dc5-85c56b59d0ab.png)

![image](https://user-images.githubusercontent.com/14789712/39845743-77797de8-53ac-11e8-8cc6-5ed48f65bffa.png)

Documentation:
![image](https://user-images.githubusercontent.com/14789712/39845782-ae1d4da2-53ac-11e8-870b-bd77865a30f5.png)


#### Checklist:
- [x] Documentation Updated
- [] Update / Add Unit Tests
- [] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

Variadicism commented on ui/packages/catalog-ui-search/src/main/webapp/component/content/alert/content.alert.less:
```diff
@@ -4,6 +4,6 @@
 
   > .content-content {
-    height: 100%;
+    height: ~'calc(100% - 1*@{minimumButtonSize})';
```
:question: What does this `1*` do in all of these `.less` files?

alexaabrd replied:
nothing - removed them all except for the right-navigation width one because that will increase as more buttons are added

alexaabrd commented:
@lcrosenbu pictures are updated with the button change and resized slideout-menu 👍

alexaabrd commented:
build now

rzwiefel commented:
Is there any way we could follow some well established trends with tab management like chrome does and add tabs like that? Having an isolated bar seems a little inefficient with our space

<img width="330" alt="screenshot 2018-04-12 17 32 18" src="https://user-images.githubusercontent.com/9013416/38710918-79d6761c-3e77-11e8-99f2-2b2576fe390a.png">

alexaabrd commented:
@rzwiefel It shouldn't be an isolated bar for long - Here is a more detailed mockup of the future ui that I think shows why it is being done this way
![image](https://user-images.githubusercontent.com/14789712/38712095-60fb5b9c-3e7e-11e8-90ae-4f588e0c78cb.png)

adimka commented:
I agree with Steve on this.
I'm not sure how users are supposed to know what this button does.
The button isn't located somewhere that gives context to it's purpose - it's not next to the visualization tabs.
The button has no label, and the icon isn't a well known icon with a singular purpose.
Even experienced users will get lost, as it's both in a new place and unlabeled.
I support the effort to move iteratively toward our new UI, but we should be conscious of our intermediate states being regressions in usability.

bdeining commented:
In both cases (the mockup and this proposed change), the add visual button is just a `+`.  How is the intermediate any different than the final goal other than the fact that there are additional buttons?  Perhaps moving the `i` button there as well in this iteration should satisfy other concerns.

TeresaHudson commented:
I agree that this button will be difficult for users to identify.  For this interim version I think we should consider at least pairing the icon with the same "Add Visual" label that they are used to.

alexaabrd commented:
* Updated to use custom icon webfont, so depends on https://github.com/codice/ddf/pull/3194
* Added 'Add Visual' Text next to the icon
* Note that once this is merged i'm going to open another PR that will add another button to the toolbar - hopefully that'll ease some concerns with this change

alexaabrd commented:
build now

bdeining commented on ui/packages/codice-icons/codice-icons/plus-sign-add.svg:
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Generator: Adobe Illustrator 22.1.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
```
Do we need this comment?

alexaabrd replied:
Nope - deleted

bdeining commented on ui/packages/catalog-ui-search/src/main/webapp/component/slideout/slideout.less:
```diff
@@ -35,6 +35,8 @@
 
 }
 
+
```
is this whitespace deliberate?

alexaabrd replied:
Nope - deleted

bdeining commented on ui/packages/catalog-ui-search/src/main/webapp/component/slideout/slideout.less:
```diff
@@ -46,23 +48,84 @@
 
-body.has-header @{customElementNamespace}slideout {
+
+
```
is this whitespace deliberate?

alexaabrd replied:
Nope - deleted

bdeining commented on ui/packages/catalog-ui-search/src/main/webapp/component/slideout/slideout.less:
```diff
@@ -46,23 +48,84 @@
-        top: ~'calc(2*@{minimumLineSize} + 1.5rem)';
-        height: ~'calc(100% - 2*@{minimumLineSize} - 3rem)';
+
+
```
is this whitespace deliberate?

alexaabrd replied:
Nope - deleted

bdeining commented on distribution/docs/src/main/resources/content/_using/using-catalog-search-ui.adoc:
```diff
@@ -263,7 +263,7 @@ Bookmarked results are denoted with a yellow *star* (image:bookmarked-icon.png[]
 Visuals are different ways to view search results.
 
-. Click the *Add Visual* (image:add-visual-icon.png[add visual icon]) icon in the bottom right corner of ${catalog-ui}.
+. Click the *[ + ]* icon in the top right corner of ${catalog-ui}.
```
Is this still true?

alexaabrd replied:
kind of - went ahead and added the image of the new button instead of the textual representation

brendan-hofmann commented:
@alexaabrd can you update the screenshots?

alexaabrd commented:
@brendan-hofmann screenshots updated ✔️

bdeining commented:
build now

alexaabrd commented:
build now

alexaabrd commented:
build now
